specification.markdown

```diff
@@ -1008,7 +1008,11 @@ If a Master Element contains more occurrences of a Child Element than permitted
 
 # Terminating Elements
 
-Null Octets, which are octets with all bits set to zero, MAY follow the value of a String Element or UTF-8 Element to serve as a terminator. An EBML Writer MAY terminate a String Element or UTF-8 Element with Null Octets in order to overwrite a stored value with a new value of lesser length while maintaining the same Element Data Size (this can prevent the need to rewrite large portions of an EBML Document); otherwise the use of Null Octets within a String Element or UTF-8 Element is NOT RECOMMENDED. An EBML Reader MUST consider the value of the String Element or UTF-8 Element to be terminated upon the first read Null Octet and MUST ignore any data following the first Null Octet within that Element. A string value and a copy of that string value terminated by one or more Null Octets are semantically equal.
+Null Octets, which are octets with all bits set to zero, MAY follow the value of a String Element or UTF-8 Element to serve as a terminator.
+An EBML Writer MAY terminate a String Element or UTF-8 Element with Null Octets in order to overwrite a stored value with a new value of lesser length while maintaining the same Element Data Size (this can prevent the need to rewrite large portions of an EBML Document); otherwise the use of Null Octets within a String Element or UTF-8 Element is NOT RECOMMENDED.
+An UTF-8 string with Null octets at the end MUST be a valid UTF-8 strings. In other words a Null octet cannot be found inside a multi-octet UTF-8 codepoint.
+An EBML Reader MUST consider the value of the String Element or UTF-8 Element to be terminated upon the first read Null Octet and MUST ignore any data following the first Null Octet within that Element.
```

Comment:
I think this means that if the reader does encounter an invalid encoding such as 0x80 0x00, it will not try to assign meaning to the "incomplete codepoint" and terminate the string after the previous fully-encoded codepoint.  It's unclear whether adding or changing the text would help clarify that or not; the previous sentence takes care of the main concern.

Reply:
Indeed, it's redundant now that we define `An UTF-8 string with Null octets at the end`. It's one or more octets. I'll remove it.